### ISSUE TITLE
Discrepancy on minimum required Python version

### ISSUE THREAD

**Issue by lvaylet** [google/slo-generator#257]
`README.md` says:

```
Requirements
- python3.9 and above
```

while `setup.py` says:

```
      classifiers=[
          release_status,
          'Intended Audience :: Developers',
          'Topic :: Software Development :: Build Tools',
          'License :: OSI Approved :: Apache Software License',
          'Programming Language :: Python',
          'Programming Language :: Python :: 3.6',
          'Programming Language :: Python :: 3.7',
          'Programming Language :: Python :: 3.8',
          'Programming Language :: Python :: 3.9',
      ],
[...]
      python_requires='>=3.4'
```

Settle on 3.9 everywhere, including in the matrix strategy in `.github/workflows/test.yml`?

**Comment by ocervell** [comment]
mmh watch out if you upgrade the minimum Python version, it might leave some users with no backward compatibility or even break their production if they've deployed manually (not sure if anyone did that yet 😃)

**Comment by lvaylet** [comment]
Good point @ocervell. That is exactly why I opened #255. `tox` works well for tests. I also wanted to cover the linting so I went for the matrix strategy approach in https://github.com/google/slo-generator/pull/256.

Now we are sure that versions 3.7 to 3.10 work as expected, including the type hints.

So let's settle on 3.7 everywhere? Support for 3.6 [ended 10 months ago](https://endoflife.date/python), on Dec 23, 2021.
